website/client-old/js/services/userServices.js

```diff
@@ -190,6 +190,8 @@ angular.module('habitrpg')
         })
       }
 
+      //Calling this multiple times rapidly will not persist changes
```

Comment:
That is a possibility, the two calls issue the setUser call, then an ajax request. If the ajax request overwrites the local version, then that would be the cause. I noticed that I could not replicate the issue while debugging with a breakpoint. If there was a delay (like a breakpoint) the calls probably had time to resolve before the next one was made. If the call could issue a promise or create a queue of requests, that would probably be the best solution. The comment might be unnecessary then.